library/src/main/java/com/google/android/exoplayer/extractor/ts/Mpeg2Reader.java

```diff
@@ -0,0 +1,189 @@
+                    samplePosition = totalBytesWritten - dataWrittenPast;
+                    timeUs = pesTimeUs; // Update the time to new pes
+                }
+                frameCollected = false;
+            }
+        }
+    }
+
+    @Override
+    public void packetFinished() {
+        // Do nothing.
+    }
+
+
+    /**
+     * Locates the next sync word, advancing the position to the byte that immediately follows it.
+     * If a sync word was not located, the position is advanced to the limit.
+     * @param data The buffer whose position should be advanced.
+     * @param startoffset The offset of the data in {@code data}.
+     * @param endoffset The limit (exclusive) of the data in {@code data}.
+     * @return True if a sync word position was found. False otherwise.
+     */
+
+    private boolean skipToNextSync(byte[] data, int startoffset, int endoffset) {
+        int length = endoffset - startoffset;
+
+        Assertions.checkState(length >= 0);
+        if (length == 0) {
+            return false;
+        }
+
+        while (startoffset < length) {
+            if((data[startoffset + 0] & 0xFF) == 0x00 && (data[startoffset + 1] & 0xFF) == 0x00 && (data[startoffset + 2] & 0xFF) == 0x01 && (data[startoffset + 3] & 0xFF) == 0xB3) {
+                seqHeaderFound = true;
+                Log.i(TAG, "Found the Sequence Header");
+                return true;
+            }
+            startoffset++;
+            continue;
+        }
+        return false;
+    }
+
+    /**
+     * Locates the next frame start code (0x00, 0x00, 0x01, 0xXX)
+     * where XX can be anything lesser than 0x01 or more than 0xaf
+     * @param data The buffer whose position should be advanced.
+     * @param startoffset The offset of the data in {@code data}.
+     * @param endoffset The limit (exclusive) of the data in {@code data}.
+     * @return The offset into the buffer where a new frame starts from.
+     */
+    private int findValidStartCode(byte[] data, int startoffset, int endoffset) {
+        int length = endoffset - startoffset;
+
+        Assertions.checkState(length >= 0);
+        if (length == 0) {
+            return endoffset;
+        }
+
+        int limit = endoffset - 3;
+        for (int i = startoffset + 3; i < limit; i += 4) {
+            if ((data[i - 1] & 0xFE) != 0) {
+                // There isn't a frame prefix here, or at the next two positions. Do nothing and let the
+                // loop advance the index by four.
+            } else if (data[i - 3] == 0 && data[i - 2] == 0 && data[i - 1] == 1) {
+                if (checkMpegVideoFrame(data[i]))
+                   return i - 3;
+            } else {
+                i -= 3;
+            }
+        }
+        return endoffset;
+    }
+
+    public  boolean checkMpegVideoFrame (int data) {
+        int byte1 = data & 0xFF;
+        if (byte1 < 0x01 || byte1 > 0xaf){
+            isSync = (byte1 == 0xb3) ? true : false;
```

Comment:
I guess it depends on what the underlying decoder expects (which I'm not sure about).